### PR TITLE
Add dump logits processor for debugging

### PR DESCRIPTION
**Pull request by will-lms**
This is only meant for debugging. It is not hooked up to anything in this PR, but can easily be registered in `generate.py`.

### CODE REVIEW

File: tests/processors/README.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# tests/processors README
+
```
Comment (neilmehta24) [inline tests/processors/README.md:2]:
```suggestion
```

File: tests/processors/dump_logits_processor.py
Comment (neilmehta24) [inline tests/processors/dump_logits_processor.py:1]: Can you import this file somewhere so we always know that the syntax and imports in here are always valid? To ensure it stays updated even if the code never runs

Comment (neilmehta24) [reply tests/processors/dump_logits_processor.py:1]: Maybe in the test `__init__.py` with a comment explaining why it's being imported
